models/Asset.php

```diff
@@ -380,9 +380,13 @@ public static function create(int $parentId, array $data = [], bool $save = true
             if (array_key_exists('type', $data)) {
                 unset($data['type']);
             }
+        } elseif (array_key_exists('type', $data)) {
+            $type = $data['type'];
+            unset($data['type']);
         }
 
-        $className = Pimcore::getContainer()->get('pimcore.class.resolver.asset')->resolve($type);
+        $className = Pimcore::getContainer()->get('pimcore.class.resolver.asset')->resolve($type)
+            ?? throw new InvalidArgumentException('Invalid asset type provided');
```

Comment:
This can happen when someone passes on a wrong `$data['type']` value. It's just throwing the exception a bit earlier, without the changes in this PR, the asset was saved to the database as it was provided (thus with a wrong type) and then you got an exception when loading the asset.

Reply:
Technically one could say this is a change of behavior. But otherwise we save something to the database that will make problems later on, the change looks good to me and can be considered a bug fix.